compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1523,6 +1531,162 @@ toCtor env con =
 isRecordCtor :: Ctor -> Bool
 isRecordCtor (Ctor _ fldNames fldTys) = not (null fldNames) || null fldTys
 
+------------------------------------------------------------------------------
+-- EXTERNAL PACKAGES
+
+-- External instance methods
+convertExternal :: Env -> LF.PackageRef -> String -> LF.Type -> LF.Expr
+convertExternal env stdlibRef primId lfType
+    | [pkgId, modStr, templName, method] <- splitOn ":" primId
```

Comment:
Why do we try to cram all information into a single string that we then have to deconstruct? It seems easier to just give `external` multiple parameters.

Reply:
I think the parameter number will be different for Choice instances and I wouldn't want to have a second primitive for those. If not, I'm happy to have a `external @pkg @mod @templ @method` primitive.

Reply:
we should be able to match on the number of parameters that `external` is applied to. That said, it doesn’t seem that unreasonable to have two primitives.